server/aws-lsp-partiql/src/server/syntax-highlighting/parser-tokens.ts

```diff
@@ -0,0 +1,174 @@
+            },
+            tokenType: tokenType,
+            tokenLength: node.text.length,
+        }
+        data.push(token)
+    }
+    return data
+}
+
+function _sortSemanticTokens(tokens: SemanticToken[]): SemanticToken[] {
+    return tokens.sort((a, b) => {
+        // Compare start line
+        if (a.range.start.line !== b.range.start.line) {
+            return a.range.start.line - b.range.start.line
+        }
+        // Compare start character
+        if (a.range.start.character !== b.range.start.character) {
+            return a.range.start.character - b.range.start.character
+        }
+        // Compare end line
+        if (a.range.end.line !== b.range.end.line) {
+            return a.range.end.line - b.range.end.line
+        }
+        // Compare end character
+        if (a.range.end.character !== b.range.end.character) {
+            return a.range.end.character - b.range.end.character
+        }
+        return 0
+    })
+}
+
+function _breakMultilineTokens(tokens: SemanticToken[]): SemanticToken[] {
+    const breakDownTokens: SemanticToken[] = []
+    for (const token of tokens) {
+        const startLine = token.range.start.line
+        const endLine = token.range.end.line
+        const startCharacter = token.range.start.character
+        const endCharacter = token.range.end.character
+        const originalLength = token.tokenLength
+        if (startLine === endLine) {
+            breakDownTokens.push(token)
+        } else {
+            const FirstToken: SemanticToken = {
+                range: {
+                    start: { line: startLine, character: startCharacter },
+                    end: { line: startLine, character: startCharacter + originalLength },
+                },
+                tokenType: token.tokenType,
+                tokenLength: originalLength,
+            }
+            breakDownTokens.push(FirstToken)
+            for (let i = startLine + 1; i < endLine; i++) {
+                const innerToken: SemanticToken = {
+                    range: {
+                        start: { line: i, character: 0 },
+                        end: { line: i, character: originalLength },
+                    },
+                    tokenType: token.tokenType,
+                    tokenLength: originalLength,
+                }
+                breakDownTokens.push(innerToken)
+            }
+            const lastToken: SemanticToken = {
+                range: {
+                    start: { line: endLine, character: 0 },
+                    end: { line: endLine, character: endCharacter },
+                },
+                tokenType: token.tokenType,
+                tokenLength: endCharacter,
+            }
+            breakDownTokens.push(lastToken)
+        }
+    }
+    return breakDownTokens
+}
+
+export function encodeSemanticTokens(
+    tokens: SemanticToken[],
+    multilineTokenSupport: boolean
+): Promise<SemanticTokens | null> {
```

Comment:
Why do you need it to return promise here?

Reply:
Thanks for catching that! It was indeed a legacy error from earlier code modifications. I've removed the promise as suggested.